packages/presentation/lib/app/app_bloc.dart

```diff
@@ -46,7 +54,15 @@ class _AppBloc extends BlocImpl implements AppBloc {
 
   @override
   void onItemTapped(int index) {
-    selectedIndex = index;
+    _selectedIndex = index;
+    switch (index) {
```

Comment:
do it via enum. so, u will have the following flow:

u will have a enum for bottom navigation
enum BottomNavigationItemType {
home
...
...
login
}

u can create a static method/extension, which consumes number as parameter and returns enum item (as each enum has its own index

u refactor your code like this

final type = BottomNavigationItemType.fromIndex(index)
switch (type) {
case BottomNavigationItemType.home -> do smth

it seems much prettier